independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java

```diff
@@ -464,9 +464,11 @@ public void run() {
                 }
             } catch (Exception e) {
                 // swallow exception and log errors for every problematic OM
-                LOGGER.errorf("Failure while notifying an observer %s for event %s. Stack trace - %s",
-                        observerMethod, eventContext.getMetadata().getType().getTypeName(),
-                        e.getCause() != null ? e.getCause() : e);
+                LOG.errorf(
+                        "Failure occurred while notifying a transational %s for event of type %s \n- please enable debug logging to see the full stack trace",
+                        observerMethod, eventContext.getMetadata().getType().getTypeName());
+                LOG.debugf(e, "Failure occurred while notifying a transational %s for event of type %s",
```

Comment:
Shouldn't this message contain the stack trace (`e.getCause()`)?

Reply:
I don't think so - we pass the exception as an argument so it's up to the logging config...